clippy_lints/src/types/mod.rs

```diff
@@ -90,7 +90,7 @@ declare_clippy_lint! {
     ///
     /// ### Why is this bad?
     /// `Option<_>` represents an optional value. `Option<Option<_>>`
-    /// represents an optional optional value which is logically the same thing as an optional
+    /// represents an optional value which is logically the same thing as an optional
```

Comment:
This change is technically, wrong, but there should be a way to clarify, how about:
```suggestion
    /// represents an optional value which itself is optional, and this is logically the same thing as an optional
```